sdx_controller/utils/db_utils.py

```diff
@@ -60,7 +60,7 @@ def read_from_db(self, collection, key):
     def get_all_entries_in_collection(self, collection):
         db_collection = self.sdxdb[collection]
         # MongoDB has an ObjectId for each item, so need to exclude the ObjectIds
-        all_entries = db_collection.find({}, {"_id": 0})
+        all_entries = db_collection.find({"deleted": {"$ne": True}}, {"_id": 0})
```

Comment:
This one just look for documents that not satisfy situation: contain a "deleted" field that is equal to True. If documents don't have `deleted` field, they are treated as "not deleted". Since this is db operation, it should usually be more efficient than implementing on our own, especially when dealing with more data.